### ISSUE TITLE
Deprecate legacy PDRFs 0-5

### ISSUE THREAD

**Issue by esilvia** [ASPRSorg/LAS#3]
*Post deleted.*
 (This thread became a discussion about deprecating the legacy PDRFs. Rather than move all of that discussion to a new thread, I instead recreated the original thread about zeroing legacy point counts in a new thread here: https://github.com/ASPRSorg/LAS/issues/12)

See the first post by @HKHeidemann for this thread's initlization.

**Comment by HKHeidemann** [comment]
I concur. 

At the risk of opening old wounds and starting a new range war, I will now renew my advocacy to drop legacy support for PDRFs 0-5 in future revisions to the LAS Specification.  Software must adapt to new format changes, not the other way round.  Every argument ever made to the contrary can easily be fixed by legacy software providers providing/including a utility to convert from  LAS v.Today  to  LAS v.Yesterday  (along with a cheerful warning about the information that is going to be lost in the process.

**Comment by hobu** [comment]
> Software must adapt to new format changes, 

Not all software gets updated (the transition to 1.4 even with the old PDRFs is much slower than you'd like @HKHeidemann), and if you're going to make them refactor all of their software to catch up to the new format, you've given them a perfect opportunity to find a better suited format. 

> drop legacy support for PDRFs 0-5

What about dropping support for *all* PDRFs and only requiring XYZ dimensions?  That's essentially what [BPF](https://nsgreg.nga.mil/doc/view?i=4202) is, minus all of LAS' prescriptions for coordinate systems and header information.

**Comment by esilvia** [comment]
I think there's a good case for replace/deprecate PDRFs in future *versions* of
LAS, but it doesn't make sense to do so in *revisions* of LAS 1.4. The
cat's already out of the proverbial bag on that issue.

Let's stay on topic, though. If we'd like to discuss this topic further,
feel free to create a new thread.

Evon

**Comment by hobu** [comment]
> The cat's already out of the proverbial bag on that issue.

Yes of course. I was talking about 2.x. Assuming semantic versioning, I don't think the format's meaning and arrangement should change so substantially in 1.x. We've chewed the cud on that discussion for a long time in the past too...

**Comment by HKHeidemann** [comment]
I was not suggesting we change LAS 1.4; apologies if I was not clear.  I
too meant LAS v2.0+





Karl

*H. Karl Heidemann, GISP, CMS*
*Physical Scientist, Lidar Science*
U.S. Geological Survey
Mundt Federal Building
47914 252nd Street
Sioux Falls, SD  57198
605-594-2861
kheidemann@usgs.gov

*"Nothing matters very much, and very few things ... matter at all."*

*- Arthur James Balfour*

**Comment by rapidlasso** [comment]
As long as we keep point types 6 to 10 I see no need to get rid of 0 through 5.  As long as the LAS format uses a *single* *fixed-sized* point record data format per file I see no compelling reason (other than to "force" users into certain point types) to deprecate any pre-existing point record. If we add new attributes to the points records we can simply add new fixed-sized point types as we have done in the past.

Point type 0 and 2 are popular with dense-matched data. Why alienate the photogrammetry community by forcing them to use (even) wider than needed point types? Not to mention the many European data portals and survey agencies, I see no rush to the wider point types in Europe (or elsewhere). While this is the *A*SPRS and not the *I*SPRS, the success of the LAS format is also due to the amount of LAS data available and the widespread support it has in software internationally.

**Comment by rapidlasso** [comment]
Karl, may I ask you to remove your signature (and the quoted message) when answering via email? Two lines of comment followed by nineteen lines of signature (plus the quoted message) make these otherwise terse github discussions really untidy.

**Comment by hobu** [comment]
> Karl, may I ask you to remove your signature

It's easy enough for us to remove it by updating the comments in the tickets. The input is more valuable, especially in the open forum.

**Comment by rapidlasso** [comment]
Cool. I did not know that was possible. (-: That's something missing in my LAStools user forum.

**Comment by hobu** [comment]
> Cool. I did not know that was possible. (-: That's something missing in my LAStools user forum.

It is something an admin can do. They can also lock tickets from further discussion once it is clear the 🐴  is 💀 .

**Comment by HKHeidemann** [comment]
each his own. Not everybody has the same needs.

Karl

*H. Karl Heidemann, GISP, CMS*
*Physical Scientist, Lidar Science*
U.S. Geological Survey
Mundt Federal Building
47914 252nd Street
Sioux Falls, SD  57198
605-594-2861
kheidemann@usgs.gov

*"Nothing matters very much, and very few things ... matter at all."*

*- Arthur James Balfour*

**Comment by esilvia** [comment]
I've changed the title of this thread to reflect its contents and recreated the original thread at #12. I concur that this issue can be addressed in the 2.x update.

**Comment by esilvia** [comment]
This is now a feature of the pending LAS 1.5 release.

**Comment by esilvia** [comment]
This is a duplicate of #128
